src/pages/home/Home.js

```diff
@@ -0,0 +1,57 @@
+/* eslint-disable react/no-array-index-key */
+import React, { useEffect, useReducer } from 'react';
+import { RowSection, Banner } from '../../components';
+import { getProducts } from '../../services/flixycartApi';
+import { productReducer } from './reducers/productReducer';
+import { loadProductActions } from '../../constants/actions';
+import './Home.scss';
+
+const Home = () => {
+  const [productState, dispatch] = useReducer(productReducer, { loading: true, product: [] });
+
+  useEffect(() => {
+    document.title = 'Flixycart | Your store for latest books';
+
+    dispatch({ type: loadProductActions.REQUEST_GET_PRODUCTS });
+
+    getProducts()
+      .then((res) => {
+        dispatch({
+          type: loadProductActions.REQUEST_GET_PRODUCTS_SUCCESS,
+          payload: res.data.result,
+        });
+      })
+      .catch((err) =>
+        dispatch({
+          type: loadProductActions.REQUEST_GET_PRODUCTS_SUCCESS,
```

Comment:
You an use here async and await to handle the aync Behavior..

Reply:
To async await,  would require another  **async function** wrapping all the code, this would cause un-necessary extra code . Hence, this way

Reply:
Ok @Debarshi95 got your point.